nemo/collections/common/tokenizers/ja_tokenizer.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+from typing import List
+
+from sacremoses import MosesDetokenizer
+
+
+class JADetokenizer:
```

Comment:
Since MosesDetokenizer() is being run for all languages anyway on L359 of `mt_enc_dec_model.py` you could undo only sentencepiece within this class?

Reply:
Do you mean instead of having a JADetokenizer, modify the MosesDetokenizer?